ProcessResultsUtils.scala

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import org.apache.spark.sql.{DataFrame, Dataset, Row, SparkSession}
+import org.slf4j.LoggerFactory
+import play.api.Configuration
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json}
+import play.api.libs.json._ // JSON library
+import play.api.libs.json.Reads._ // Custom validation helpers
+import play.api.libs.functional.syntax._ // Combinator syntax
+
+object ProcessResultsUtils extends Serializable {
+  private val LOGGER = LoggerFactory.getLogger("ProcessResultsUtils")
+  var debugMode = false
+
+  def setDebugMode (debug: Boolean) : Unit = {
+    debugMode = debug;
+  }
+
+  def anyNotNull(row: Row): Boolean = {
+    val len = row.length
+
+    var i = 0
+    var fieldNames = row.schema.fieldNames
+    //print patientId if its the only col
+    if (len == 1 && fieldNames(0).equals(Constants.SUBJECT_ID_COL_NAME))
+      return true
+    //skip patientId
+    while (i < len) {
+      if (!fieldNames(i).equals(Constants.SUBJECT_ID_COL_NAME) && !row.isNullAt(i)) {
+        return true
+      }
+      i += 1
+    }
+    false
+  }
+
+
+  def loadTableDFFromConfig(tableFilePrefix : String, spark: SparkSession, config: Configuration,
+                            dataConfigName: String): DataFrame = {
+    LOGGER.info("PloadTableDFFromConfig")
+
+    val connInfo = config.get[String](dataConfigName)
+    if (connInfo.contains("s3a")) {
+      var dataDF: DataFrame = null
+
+      dataDF = spark.read.parquet(connInfo)
+      return dataDF
+    }
+    //Use jdbc connection:
+    val url = config.get[String]("db.mysql.url")
+    val user = config.get[String]("db.mysql.username")
+    val pass = config.get[String]("db.mysql.password")
+    var jdbcDF = spark.read.format("jdbc").option("url", url).option("dbtable", connInfo).
+      option("user", user).option("password", pass).load
+    return jdbcDF
+  }
+
+
+  def addTableToSpark (spark: SparkSession, config: Configuration,
+                       dataConfigName: String) : Unit = {
+    LOGGER.info("addTableToSpark")
+    var tableDF = loadTableDFFromConfig(null, spark, config, dataConfigName)
+    var sparkName = dataConfigName.toString()
+    if (dataConfigName.toString().contains("clauses")) {
+      sparkName = "clauses"
+    }
+    tableDF.createOrReplaceTempView(sparkName)
+    if (debugMode) {
+      println("============= " + sparkName + " ===============")
+      tableDF.show(false)
+    }
+  }
+
+  def createJoinDataFrame (spark: SparkSession, query:String, config: Configuration, testType: String): Unit = {
```

Comment:
This method shouldn't be the one parsing the enforcement engine response. You have to create a structure that holds the tables and the query, and this method should create the join data frame based on them.